flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/collect/CollectSinkOperatorCoordinator.java

```diff
@@ -152,9 +176,11 @@ private CoordinationResponse handleRequestImpl(
             throw new NullPointerException("No sinkAddress available.");
         }
 
-        if (socketConnection == null) {
-            socketConnection = SocketConnection.create(socketTimeout, sinkAddress);
-            LOG.info("Sink connection established");
+        synchronized (connectionLock) {
```

Comment:
Wondering if this shouldn't be a larger block.
Does the client handle it properly when the connection breaks down while it is reading data?

What's the desired behavior here?

Reply:
I didn't consider this one because of the [findings below](https://github.com/apache/flink/pull/23296#discussion_r1307474170)